src/service-override/views.ts

```diff
@@ -304,16 +334,45 @@ function registerEditorPane (options: EditorPanelOption): { disposable: IDisposa
     }
   }
 
-  const disposable = Registry.as<IEditorPaneRegistry>(EditorExtensions.EditorPane).registerEditorPane(
+  const disposableStore = new DisposableStore()
+
+  disposableStore.add(Registry.as<IEditorPaneRegistry>(EditorExtensions.EditorPane).registerEditorPane(
     EditorPaneDescriptor.create(
       CustomEditorPane,
       options.id,
       options.name
     ),
-    [new SyncDescriptor(CustomEditorInput)])
+    [new SyncDescriptor(CustomEditorInput)]))
+
+  const registerEditorOptions = options.registerEditorOptions
+  if (registerEditorOptions != null) {
+    getService(IEditorResolverService).then(editorResolverService => {
+      if (disposableStore.isDisposed) {
+        return
+      }
+
+      disposableStore.add(editorResolverService.registerEditor(
+        registerEditorOptions.globPattern,
+        {
+          id: CustomEditorInput.ID,
+          label: options.name,
+          priority: registerEditorOptions.priority ?? RegisteredEditorPriority.default
+        },
+        {},
+        {
+          createEditorInput (editorInput: IResourceEditorInput | ITextResourceEditorInput, _group: IEditorGroup) {
+            return {
+              options: {},
+              editor: new CustomEditorInput(undefined, editorInput)
+            }
+          }
+        }
+      ))
+    }).catch(console.error)
```

Comment:
I'm not really happy with having an async method the caller is not able to await nor catch errors

Instead, what about returning an async `registerEditor(globPattern: string, options?: Partial<Omit<RegisteredEditorInfo, 'id'>>)` method, next to the `disposable` and `CustomEditorInput`? That would also allow to register multiple glob patterns for a single editorPane

What do you think?